### PR TITLE
support for more fieldTypes and generating unique ids for the fields

### PR DESCRIPTION
**Pull request by vdua**
## depends on 

#3 

## Test URLs:
- Before: https://form-block1--nedbank--vdua.hlx.live/en/personal/borrow/personal-loans-form
- After: https://form-block2--nedbank--vdua.hlx.live/en/personal/borrow/personal-loans-form

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -24,44 +51,69 @@ function createButton(fd) {
...
 }
 
 function createInput(fd) {
   const input = document.createElement('input');
   input.type = fd.Type;
-  input.id = fd.Field;
-  input.setAttribute('placeholder', fd.Placeholder);
-  if (fd.Mandatory === 'x') {
+  input.id = fd.Id;
+  if (fd.Mandatory === 'TRUE') {
```
Comment (trieloff) [inline blocks/form/form.js:64]: Ignore case.

File: blocks/form/form.js [hunk 2]
```diff
@@ -73,24 +125,30 @@ function renderField(fd) {
...
   const json = await resp.json();
-  return json;
+  return json.data.map((fd) => ({
+    ...fd,
+    Id: fd.Id || getId(fd.Name),
+  }));
 }
 
-async function fetchForm(pathname) {
+async function fetchForm(pathname, getId) {
```
Comment (trieloff) [inline blocks/form/form.js:143]: passing `idGenerator` through is strange, when you can just use the function as a singleton and put the map of ids into a closure.

**Review by trieloff — COMMENTED** [review]
Let's re-review this after the prerequisite has been merged.

File: blocks/form/form.js (hunk 2 above)
Comment (vdua) [reply blocks/form/form.js:143]: I agree. Not sure what i was thinking

Comment (vdua) [reply blocks/form/form.js:143]: resolved.
https://github.com/vdua/nedbank/pull/1/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R162

File: blocks/form/form.js (hunk 1 above)
Comment (vdua) [reply blocks/form/form.js:64]: resolved
https://github.com/vdua/nedbank/pull/1/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R221

**Comment by vdua** [comment]
The review comments have been incorporated in #1 itself.

**Pull request closed without merging**